pkg/networkutils/networkutils.go

```diff
@@ -42,3 +42,13 @@ func IsIPInUse(client NetClient, ip string) bool {
 
 	return false
 }
+
+func GetLocalIP(client NetClient) (net.IP, error) {
```

Comment:
I don't think there's a need to inject the `NetClient`. I appreciate the testability concern but this isn't the API consumers really want. They really want a `GetLocalIP()` and thats all. We should probably embrace that - as should the IP checking function at some point - and do the best we can with testing. 

This also means we can get rid of the timeout and simplify to just `net.Dial(...)`.